#!/usr/bin/env python
PACKAGE = 'goal_manager'

from dynamic_reconfigure.parameter_generator_catkin import *

gen = ParameterGenerator()

gen.add('x_position',     double_t,   0, 'frame_goal_x in [m]', 0.0, 0, 1.7)
gen.add('y_position',     double_t,   0, 'frame_goal_y in [m]', 0.0, 0, 1.5)
gen.add('z_position',     double_t,   0, 'frame_goal_z in [m]', 0.0, 0, 1.2)

gen.add('waittime',   double_t,   0, 'Timeout at reached goalpoint [s]', 1.0, 0, 5)
gen.add('fixed_frame',              str_t,      0, 'Fixed Frame', 'odom')
gen.add('rabbit_frame',              str_t,      0, 'Fixed Frame', 'rabbit')
gen.add('robot_frame',              str_t,      0, 'Fixed Frame', 'base_link')

exit(gen.generate(PACKAGE, 'goal_manager', 'FrameGoalManager'))
